### DISCUSSION TITLE
How do you loop over all nodes to alter the solution vector

### DISCUSSION THREAD

**Discussion by starkekr** [mfem/mfem#4220] [General]
I want to loop over all nodes in the simulation and alter the solution vector. Right now I have the following code from a previous discussion:
const Operator &P = *fespace.GetProlongationMatrix();
Vector x_nodal(P.Height());
P.Mult(X,x_nodal);
for (int i = 0; i < x_nodal.Size(); i++)
{
// I want to get the nodal position here but don't know where to get this information.
}
const Operator &R = *fespace.GetRestrictionMatrix();
R.Mult(x_nodal, X);

 Is there a dof->node map on the mesh object that I can use to get the positions of the nodes for a particular dof?

**Comment by vladotomov** [comment]
@starkekr see `FiniteElement::GetNodes()`. But you'll have to iterate over mesh elements.

Another option is `Mesh::GetGeometricFactors` which can give you a single `Vector` of all coordinates for an `IntegrationRule`. The rule can be created again by `FiniteElement::GetNodes()`. You'll have to be careful with the ordering.
